### ISSUE TITLE
vdwr 2 - clash score improvement vs. magnesium bumping

### ISSUE THREAD

**Issue by helgepat** [keitaroyam/servalcat#10]
Hi again,

I have a question which I think is mostly about Refmac, but it would be nice to hear your input on the issue. Also is there a place to ask Refmac-related questions directly?

I tried out the "vdwr 2" keyword through servalcat and it has improved my model statistics significantly, but now I am skeptical if this is ok to use for my model (map resolution ~2A).

With our ribosome maps, often the exterior has weaker density than the core and I think thats were a lot of the clashes originate. So running "vdwr 2" seems to prevent the exterior chains from clashing which is nice... but I am wondering what the caveats are? For example, I observed for one magnesium-water cluster in the core with good map density that the distances of the waters + coordinating oxygens to the magnesium ion are ~2A before refinement (modelled in Coot), but deviate from this pattern more and more after refinement the higher I set the "vdwr" value. Is this an indication that the increased "bumpiness" is pushing my magnesium away from the optimal position? Is there something that can be done to keep this from happening while keeping the benefits of the increased vdwr setting?

**Comment by keitaroyam** [comment]
You can ask Refmac related questions here, particularly for cryo-EM structure refinement. Other place would be CCP4BB (or contact Garib directly).

What was actual Mg-O distance with high vdwr value (and what was it)? Did they move out of density?

There are two possible scenarios for Mg-O atom pairs:
1. if link is defined in the input (LINK record in PDB header or _struct_conn in mmCIF), bond distance restraint is imposed with ideal value of 1.99±0.02 Å.
2. otherwise, there will be antibumping force if Mg-O distance is shorter than 1.99 Å.

(actually, this 1.99 Å is a bit shorter than it should be)

So in case 2, Mg-O distance can be >2 Å (as there is no attracting force), but it is not because of vdwr, unless there is other atom nearby.

**Comment by helgepat** [comment]
Here are the distances of the starting model (done in Coot):
![1_starting_model](https://user-images.githubusercontent.com/97705752/212465130-3b9a2c8a-9392-414d-bf47-b70f40d978ad.png)

Refinement with "vdwr 1":

![2_vdwr1](https://user-images.githubusercontent.com/97705752/212465143-38e8fbdb-e8a2-4dc0-bc59-719afdfb680d.png)


Refinement with "vdwr 1.5":
![3_vdwr1-5](https://user-images.githubusercontent.com/97705752/212465150-a620d26b-387d-4f5e-9c5e-3fc5d000a8d5.png)

Refinement with "vdwr 2":
![4_vdwr2](https://user-images.githubusercontent.com/97705752/212465153-a601e069-cbf6-4215-9a17-5df299181847.png)


Comparison with density:
![withdensity](https://user-images.githubusercontent.com/97705752/212465157-ea8ffbf4-ef50-40fe-8699-5474d50ea5b2.png)

**Comment by helgepat** [comment]
What is the best way to add the LINK records for all magnesium atoms (and probably also potassium?). Can this be done automatically? For the Zink atoms I see that there are some added by servalcat/refmac.
Is restraining the magnesium distances something that should be avoided, or is it an accepted practice? Also coming back to the "vdwr" setting: Are there some caveats for using it in refinement in our case, or would you say its safe to use even for high-resoluton maps?

**Comment by helgepat** [comment]
I tried adding links manually in Coot and also by running Readyset from Phenix and then one mock-refinement in Phenix to write the links to the .pdb file.

When I run servalcat afterwards on these models, Refmac restrains the magnesium distances. There might be one caveat here though: I observe some water molecules that are not at  ~2A from the magnesium, but at 2.2A - 2.3A. There is clear density for the water, so I am thinking these should really be a bit further out:

Model before Servalcat:
![before](https://user-images.githubusercontent.com/97705752/212569120-b16c8ed7-225c-4e18-a23d-fef61841c23c.png)





Model after Servalcat:
![after](https://user-images.githubusercontent.com/97705752/212569121-671462e3-1b76-4ab7-acb0-04d5063a6dd5.png)



I searched a bit through the CCP4BB and I found some mention that the magnesium-oxygen distance ranges from 2.0-2.2A, is Refmac currently only setting this to 2.0A in refinement?

**Comment by keitaroyam** [comment]
Yes, Refmac's default Mg-O distance of 1.99±0.02 Å is a bit too short and tight. Actually we have been working on improved metal non-metal distance restraints in servalcat side, but it's not done (it's a good reminder!). A quick solution to this problem would be to write a dictionary.

Save the following text as e.g. mg_o.cif and give it with --ligand option. Sigma value of 0.1 Å may be too large and you may need to play with it. I only defined MG-HOH and MG-G links as I could only see them from your figure. Please add if other residues are involved (if there are many, I can write a script).

```
data_link_list
loop_
_chem_link.id
_chem_link.comp_id_1
_chem_link.mod_id_1
_chem_link.group_comp_1
_chem_link.comp_id_2
_chem_link.mod_id_2
_chem_link.group_comp_2
_chem_link.name
GMG1    G . . MG . . .
GMG2    G . . MG . . .
MGHOH HOH . . MG . . .

data_link_GMG1
loop_
    _chem_link_bond.link_id
    _chem_link_bond.atom_1_comp_id
    _chem_link_bond.atom_id_1
    _chem_link_bond.atom_2_comp_id
    _chem_link_bond.atom_id_2
    _chem_link_bond.type
    _chem_link_bond.value_dist
    _chem_link_bond.value_dist_esd
GMG1      1      OP1  2     MG    .             2.100        0.100

data_link_GMG2
loop_
    _chem_link_bond.link_id
    _chem_link_bond.atom_1_comp_id
    _chem_link_bond.atom_id_1
    _chem_link_bond.atom_2_comp_id
    _chem_link_bond.atom_id_2
    _chem_link_bond.type
    _chem_link_bond.value_dist
    _chem_link_bond.value_dist_esd
GMG2      1      OP2  2     MG    .             2.100        0.100

data_link_MGHOH
loop_
    _chem_link_bond.link_id
    _chem_link_bond.atom_1_comp_id
    _chem_link_bond.atom_id_1
    _chem_link_bond.atom_2_comp_id
    _chem_link_bond.atom_id_2
    _chem_link_bond.type
    _chem_link_bond.value_dist
    _chem_link_bond.value_dist_esd
MGHOH      1      O  2     MG    .             2.100        0.100
```

I think `vdwr 2` can be safely used at 2 Å, and Refmac's default sigma of 0.2 Å for vdwr is a bit too soft (will be 0.05 Å by vdwr 2). But there may be some problematic cases - please let me know if you find it.

**Comment by helgepat** [comment]
Thats good to hear that you are already working on it! Are you planning on adding the LINK records for these coordinations to the output.pdb as well? For the time being, I will use Phenix to add those, as described above?

I will try out the dictionary, but we probably have additional interactions in our models as we have ~250-300 Mg2+ as well as ~100-125 K+ in our ribosomes. The Zn atoms seem to be recognized as there are specific LINKR records being added after servalcat btw.

How would you define the "problemtatic cases" that I should look out for ;)

And when you say the sigma is too soft, should I submit an increased value to the keyword for servalcat, like `"vdwr 2 SIGMa VDW xxx"`? BTW: What does the Sigma Metal do here?

**Comment by keitaroyam** [comment]
I think if you use the dictionary cif file I pasted above, it will automatically detect MG-G and MG-HOH links (in servalcat), so you do not have to add LINK records manually. LINKs will be added in the output file as well.

Zn links may be added, if the particular links are defined in the monomer library. Refmac's K-O distance seems to be 2.79 A, which looks ok (right?). In this case just having LINK records would be sufficient (so it's good if the phenix program adds LINK records to all K-O bonds). So the problem is other MG distances. Let me prepare the script. In the meantime, can you try the cif file and check the result (to see if sigma of 0.1 A was ok)?

"Problematic cases": please just do what you usually do - like looking at model and density. I mean, I do not have any particular possible problem in my mind.. There may be inappropriate critical vdwr distance that we did not notice with default sigma.

Regarding vdwr keyword, the default sigma is 0.2 A, and "vdwr 2" makes it 0.2/4=0.05 A (equivalent). Because the number after vdwr is squared. Sigma metal is to change default sigma value for metal involving nonbonded interactions, but it's irrelevant if LINK is defined.

**Comment by keitaroyam** [comment]
I wrote a script: [make_mg_dict.zip](https://github.com/keitaroyam/servalcat/files/10427014/make_mg_dict.zip). I also included the result from:
```sh
python make_mg_dict.py 7k00.cif 2.1 0.1
```
Please use python that servalcat is using, and give model file name, ideal distance, sigma value. It writes a cif file that ends with _mg_links.cif, which can be used with --ligand option in servalcat. You can also add LINK records based on this cif file with
```sh
servalcat util fix_link 7k00.cif --ligand 7k00_mg_links.cif
```

**Comment by keitaroyam** [comment]
Servalcat 0.4.24 has been released, which can use new metal ideal distances from monomer library https://github.com/MonomerLibrary/monomers/pull/28. It should solve this Mg-O distance problem. The new monomer library will be available from upcoming CCP4 8.0.012.
